GetStream/stream-chat-android
# [CAS-1269] Refactor `ChannelListViewModel` to use OP based `queryChannels`

https://stream-io.atlassian.net/browse/CAS-1269

### 🎯 Goal

To merge ChatDomain with LLC, We need to replace the usage of QueryChannelsController with OfflinePlugin (ChatClient::referenced::queryChannels) and move preconditions logic to the plugin.

### 🛠 Implementation details
- A precondition function `onQueryChannelsPrecondition` added to `QueryChannelsListener`.
- `ChannelListViewModel` uses `ChatClient::queryChannels` if OfflinePlugin is enabled.

### 🧪 Testing

- Run Compose and UI sample app.
- Channel list should be loaded as earlier.
- Try enabling/disabling the offline plugin and you should not notice any difference.

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist
